src/Directory.Build.props

```diff
@@ -0,0 +1,19 @@
+<Project>
+  <PropertyGroup>
+    <LangVersion>10.0</LangVersion>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <Company>MongoDB Inc.</Company>
+    <Copyright>Copyright © 2023-present MongoDB Inc.</Copyright>
+    <Authors>MongoDB Inc.</Authors>
+    <PackageIcon>packageIcon.png</PackageIcon>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
```

Comment:
Do we want to move rest of the properties here, like
PackageTags
PackageRequireLicenseAcceptance
PackageLanguage
...